src/azure-cli-core/azure/cli/core/__init__.py

```diff
@@ -145,12 +145,13 @@ def _update_command_table_from_modules(args):
                          "(note: there's always an overhead with the first module loaded)",
                          cumulative_elapsed_time)
 
-        def _update_command_table_from_extensions():
-
-            extensions = get_extension_names()
+        def _update_command_table_from_extensions(ext_filters):
+            extensions = get_extensions()
             if extensions:
-                logger.debug("Found %s extensions: %s", len(extensions), extensions)
-                for ext_name in extensions:
+                logger.debug("Found %s extensions: %s", len(extensions), [e.name for e in extensions])
+                allowed_extensions = list(six.moves.filterfalse(lambda e: any(_wrap_block_extension_func(f, e)
+                                                                              for f in ext_filters), extensions))
```

Comment:
list comprehension would be clearer here

Reply:
I've expanded this out to a full function instead of using `filterfalse` and `any`.
Hopefully it should make things clearer.